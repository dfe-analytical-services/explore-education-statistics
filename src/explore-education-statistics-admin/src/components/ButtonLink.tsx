import classNames from 'classnames';
import React, { ReactNode } from 'react';
import { Link as RouterLink, LinkProps } from 'react-router-dom';

import {
  SetCommonButtonLink,
  ButtonLinkType,
} from '@common/components/ButtonLink';

type Props = {
  children: ReactNode;
  className?: string;
  disabled?: boolean;
  variant?: 'secondary' | 'warning';
} & LinkProps;

const ButtonLink = ({
  children,
  className,
  disabled = false,
  to,
  variant,
  ...props
}: Props) => {
  return (
    <RouterLink
      {...props}
      to={to}
      className={classNames(
        'govuk-button',
        {
          'govuk-button--disabled': disabled,
          'govuk-button--secondary': variant === 'secondary',
          'govuk-button--warning': variant === 'warning',
        },
        className,
      )}
      aria-disabled={disabled}
    >
      {children}
    </RouterLink>
  );
};

SetCommonButtonLink(ButtonLink as ButtonLinkType);

export default ButtonLink;
