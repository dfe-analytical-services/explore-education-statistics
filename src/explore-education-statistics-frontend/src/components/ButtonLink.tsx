import classNames from 'classnames';
import Link from 'next/link';
import React from 'react';
import {
  SetCommonButtonLink,
  ButtonLinkType,
} from '@common/components/ButtonLink';
import { LinkProps } from './Link';

type Props = {
  disabled?: boolean;
} & LinkProps;

const ButtonLink = ({
  children,
  className,
  disabled = false,
  to,
  prefetch,
  ...props
}: Props) => {
  const isAbsolute = typeof to === 'string' && to.startsWith('http');

  const link = (
    <a
      {...props}
      href={isAbsolute ? (to as string) : undefined}
      className={classNames(
        'govuk-button',
        {
          'govuk-button--disabled': disabled,
        },
        className,
      )}
      aria-disabled={disabled}
    >
      {children}
    </a>
  );

  if (isAbsolute) {
    return link;
  }

  return (
    <Link {...props} prefetch={prefetch} href={to} passHref>
      {link}
    </Link>
  );
};

SetCommonButtonLink(ButtonLink as ButtonLinkType);

export default ButtonLink;
