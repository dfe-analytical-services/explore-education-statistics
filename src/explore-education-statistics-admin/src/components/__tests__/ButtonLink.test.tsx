import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import React from 'react';
import ButtonLink from '../ButtonLink';

describe('ButtonLink', () => {
  test('renders correctly with required props', () => {
    const { container } = render(
      <MemoryRouter>
        <ButtonLink to="/">Test button</ButtonLink>
      </MemoryRouter>,
    );

    expect(container.innerHTML).toMatchSnapshot();
  });

  test('correctly set disabled attributes', () => {
    const { getByText } = render(
      <MemoryRouter>
        <ButtonLink to="/">Test button</ButtonLink>
      </MemoryRouter>,
    );

    const button = getByText('Test button');

    expect(button).not.toHaveAttribute('disabled');
    expect(button).toHaveAttribute('aria-disabled', 'false');
  });
});
