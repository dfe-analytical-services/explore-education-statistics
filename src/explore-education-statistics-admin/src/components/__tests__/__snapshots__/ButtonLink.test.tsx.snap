// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLink renders correctly with required props 1`] = `
<a class="govuk-button"
   aria-disabled="false"
   href="/"
>
  Test button
</a>
`;
